infra/modules/database/main.tf

```diff
@@ -40,36 +84,47 @@ resource "aws_security_group" "database" {
 ############################
 ## Database Configuration ##
 ############################
-resource "aws_rds_cluster" "postgresql" {
+resource "aws_rds_cluster" "database" {
   # checkov:skip=CKV2_AWS_27:have concerns about sensitive data in logs; want better way to get this information
   # checkov:skip=CKV2_AWS_8:TODO add backup selection plan using tags
   # checkov:skip=CKV_AWS_313: This is literally a new check; more research needed
+  # checkov:skip=CKV_AWS_324: This is literally a new check; more research needed
+  # checkov:skip=CKV_AWS_325: This is literally a new check; more research needed
+  # checkov:skip=CKV_AWS_327: This is literally a new check; more research needed
+
   cluster_identifier                  = var.database_name
-  engine                              = "aurora-postgresql"
+  engine                              = "aurora-${var.database_type}"
   engine_mode                         = "provisioned"
+  engine_version                      = var.database_type == "postgresql" ? "14.6" : "8.0.mysql_aurora.3.02.0"
```

Comment:
These engine versions are _so_ confounding

Reply:
Yah this took me ages. The documentation is really poor